File: Ruby/Basico/calculadora_cli/calculadora.rb
def calcular(a, op, b)
  case op
  when '+' then a + b
  when '-' then a - b
  when '*' then a * b
  when '/' then b != 0 ? a / b.to_f : 'Error: división por cero'
  else 'Operador no válido'
  end
end

puts '=== Calculadora CLI ==='
loop do
  print 'Operación (ej: 5 + 3) o "salir": '
  entrada = gets.chomp
  break if entrada == 'salir'

  partes = entrada.split
  if partes.length == 3
    a, op, b = partes[0].to_f, partes[1], partes[2].to_f
    puts "Resultado: #{calcular(a, op, b)}"
  else
    puts 'Formato inválido. Usá: número operador número'
  end
end

File: PHP/Basico/calculadora_cli/calculadora.php
<?php

function calcular(float $a, string $op, float $b): string|float
{
    return match ($op) {
        '+' => $a + $b,
        '-' => $a - $b,
        '*' => $a * $b,
        '/' => $b != 0 ? $a / $b : 'Error: división por cero',
        default => 'Operador no válido',
    };
}

echo "=== Calculadora CLI ===\n";

while (true) {
    echo "Operación (ej: 5 + 3) o 'salir': ";
    $entrada = trim(fgets(STDIN));

    if ($entrada === 'salir') break;

    $partes = explode(' ', $entrada);
    if (count($partes) === 3) {
        [$a, $op, $b] = $partes;
        echo "Resultado: " . calcular((float)$a, $op, (float)$b) . "\n";
    } else {
        echo "Formato inválido. Usá: número operador número\n";
    }
}
